src/library_wasmfs.js

```diff
@@ -95,11 +116,18 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       mode = typeof mode == 'undefined' ? 438 /* 0666 */ : mode;
       return withStackSave(() => {
         var buffer = stringToUTF8OnStack(path);
-        return __wasmfs_open({{{ to64('buffer') }}}, flags, mode);
+        var fd = __wasmfs_open({{{ to64('buffer') }}}, flags, mode);
+        FS.handleError(fd);
+        return fd;
```

Comment:
Reading this code, I think it could be
```js
        return FS.handleError(__wasmfs_open({{{ to64('buffer') }}}, flags, mode));
```
That would work if `handleError` returns the input if it doesn't throw. I think that's worth doing since it makes the code 1 line instead of 3 in each place.